kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -449,7 +458,10 @@ private static void validateSchemaEvolution(
    * fields
    */
   private static void validateSchemaEvolutionById(
-      StructType currentSchema, StructType newSchema, Set<String> clusteringColumnPhysicalNames) {
+      StructType currentSchema,
+      StructType newSchema,
+      Set<String> clusteringColumnPhysicalNames,
+      Map<String, String> configuration) {
```

Comment:
nit - I think I'd prefer this to be more specific for now, until in the future we maybe need this for more things. Can the arg instead just be `boolean icebergWriterCompatV1Enabled`?